A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js

```diff
@@ -137,34 +138,32 @@ class CDUAocOfpData {
         };
         mcdu.onLeftInput[1] = (value) => {
             if (value === FMCMainDisplay.clrValue) {
-                mcdu.aocWeight.taxiFuel = "";
+                mcdu.aocWeight.taxiFuel = undefined;
                 updateView();
                 return true;
             }
-            const maxAllowableFuel = 21273;
-            const enteredFuel = Math.round(+value);
+            const enteredFuel = Math.round(+value) / mcdu._conversionWeight;
             if (enteredFuel >= 0 && enteredFuel <= maxAllowableFuel) {
-                mcdu.aocWeight.taxiFuel = enteredFuel.toString();
+                mcdu.aocWeight.taxiFuel = enteredFuel;
                 updateView();
                 return true;
             }
             mcdu.addNewMessage(NXSystemMessages.notAllowed);
             return false;
         };
 
-        mcdu.rightInputDelay[1] = () => {
+        mcdu.leftInputDelay[2] = () => {
             return mcdu.getDelayBasic();
         };
-        mcdu.onRightInput[1] = (value) => {
+        mcdu.onLeftInput[2] = (value) => {
             if (value === FMCMainDisplay.clrValue) {
-                mcdu.aocWeight.tripFuel = "";
+                mcdu.aocWeight.tripFuel = undefined;
                 updateView();
                 return true;
             }
-            const maxAllowableFuel = 21273;
-            const enteredFuel = Math.round(+value);
+            const enteredFuel = Math.round(+value) / mcdu._conversionWeight;
             if (enteredFuel >= 0 && enteredFuel <= maxAllowableFuel) {
```

Comment:
isn't `maxAllowableFuel` undefined at this stage?

Reply:
My understanding was that arrow functions 'inherit' their parent's scope (see line 178 where we access `currentBlockFuel` for instance): https://medium.com/@iampika/es6-arrow-functions-syntax-and-lexical-scoping-d061732071e7

However, I'm not a Javascript expert so I might be misunderstanding something. If we're not sure if it's safe I can certainly revert this.

Reply:
I see, I am not an JS expert either, so I'll wait for @beheh response.

Reply:
Yeah, it's inherited downwards, so this code is fine.